github-bot: update deploy procedure after default branch change

phillipj opened the pull request:
The nodejs/github-bot' default branch just changed to `main`. That also means our deploy procedure has to change accordingly, to listen for commits and pull from `main` as opposed to `master`.

Refs https://github.com/nodejs/github-bot/issues/302

mhdawson reviewed (approved):
LGTM

mhdawson commented:
@phillipj does this need to be run after this lands or have the changes already been made by hand and this is to make sure we do the right thing the next time we run ansible?

phillipj commented:
@mhdawson I refrained from doing these changes by hand.

I can give ansible a try, even tho I had some local issues reading secrets the last time I tried..

phillipj commented:
Decrypting secrets worked this time around, but I'm banging my head against the wall running ansible. Getting these warnings:

```
$ ansible infra-rackspace-debian8-x64-1 -m ping -vvvv

ansible [core 2.11.2]
  config file = ~/../build/ansible/ansible.cfg
  configured module search path = [u'~/../build/ansible/plugins/library']
  ansible python module location = ~/Library/Python/2.7/lib/python/site-packages/ansible
  ansible collection location = ~/.ansible/collections:/usr/share/ansible/collections
  executable location = ~/Library/Python/2.7/bin/ansible
  python version = 2.7.16 (default, Nov 23 2020, 08:01:20) [GCC Apple LLVM 12.0.0 (clang-1200.0.30.4) [+internal-os, ptrauth-isa=sign+stri
  jinja version = 2.10
  libyaml = True

...

[WARNING]:  * Failed to parse /../build/ansible/plugins/inventory/nodejs_yaml.py with script plugin: Inventory script
(/../build/ansible/plugins/inventory/nodejs_yaml.py) had an execution error: Traceback (most recent call last):   File
"/../build/ansible/plugins/inventory/nodejs_yaml.py", line 34, in <module>     import yaml   File "/Library/Python/2.7/site-packages/yaml/__init__.py", line 2, in
<module>     from error import * ModuleNotFoundError: No module named 'error'
  File "/Users/flipp/Library/Python/2.7/lib/python/site-packages/ansible/inventory/manager.py", line 290, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/Users/flipp/Library/Python/2.7/lib/python/site-packages/ansible/plugins/inventory/script.py", line 169, in parse
    raise AnsibleParserError(to_native(e))
[WARNING]:  * Failed to parse /../build/ansible/plugins/inventory/nodejs_yaml.py with ini plugin:
/../build/ansible/plugins/inventory/nodejs_yaml.py:25: Expected key=value host variable assignment, got: __future__
  File "/Users/flipp/Library/Python/2.7/lib/python/site-packages/ansible/inventory/manager.py", line 290, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/Users/flipp/Library/Python/2.7/lib/python/site-packages/ansible/plugins/inventory/ini.py", line 136, in parse
    raise AnsibleParserError(e)
[WARNING]: Unable to parse /../build/ansible/plugins/inventory/nodejs_yaml.py as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: Could not match supplied host pattern, ignoring: infra-rackspace-debian8-x64-1
```

I've tried reinstalling ansible more than once, but can't get this to work at all. Would appreciate any protips as being an ansible rookie, I'm out of options.

richardlau commented:
@phillipj Which directory are you in when running ansible? I was talking to @AshCripps this morning and it looks like our playbooks expect to be run from the `ansible` directory of the build repository. The `/../build/ansible/plugins/inventory/nodejs_yaml.py` path looks odd.

Otherwise it might be a Python 2 incompatibility.

AshCripps commented:
Also I recently changed the inventory parser to be python3 _only_ so if you are running python2 that could be a problem

phillipj commented:
Thanks for the much needed hints, upgrading to python v3.9 did the trick!

```
TASK [package-upgrade : upgrade installed packages] *************************************************************************************************************************************
changed: [infra-rackspace-debian8-x64-1]

TASK [github-bot : Deploy Webhook | Copy config] ****************************************************************************************************************************************
changed: [infra-rackspace-debian8-x64-1]

TASK [github-bot : Deploy Webhook | Copy deploy script] *********************************************************************************************************************************
changed: [infra-rackspace-debian8-x64-1]

PLAY RECAP ******************************************************************************************************************************************************************************
infra-rackspace-debian8-x64-1 : ok=30   changed=3    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0
```

Merging since these changes has been applied.
